tests/Doctrine/fixtures/expected_no_overwrite/src/Entity/UserAvatar.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Symfony\Bundle\MakerBundle\Tests\Doctrine\fixtures\source_project\src\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity()
+ */
+class UserAvatar
+{
+    /**
+     * @ORM\Id
+     * @ORM\GeneratedValue
+     * @ORM\Column(type="integer")
+     */
+    private $id;
+
+    /**
+     * @ORM\ManyToOne(targetEntity="User", inversedBy="avatars", cascade={"persist", "remove"})
+     * @ORM\JoinColumn(nullable=false)
+     */
+    private $user;
+
+    public function getId(): ?int
+    {
+        return $this->id;
+    }
+
+    public function getUser(): ?User
+    {
+        return $this->user;
+    }
+
+    public function setUser(?User $user)
+    {
+        $this->user = $user;
```

Comment:
What if I update this directly? Then `User#avatars` will not be updated

Reply:
Do you mean: if you call $avatar->setUser($user), the problem is that the avatar will not all be added to User#avatars? i.e. the problem is that the relationship is not synced?

If so, it’s by design. If you set the inverse side of a relation, the generated code *does* sync the opening side. But not the other way around. The reason is that (a) it doesn’t need to be set for persistence and (b) more importantly, it could have major performance consequences. For example, if there are 1000 avatar, by setting User#avatars, Doctrine would first query for all 1000 and then set it. We gotta avoid that :)

Reply:
Here's an example:

1. Create a new article, `setCategory($category)`.
2. Something later (eg. listener) grabs `$article->getCategory()` and passes it somewhere else
3. It ends up in code that does `foreach ($category->getArticles() as $article)`
4. Weird results

I understand what you are saying but relying on chance that the collection will not be used later during request.... not really sound strategy.

> Doctrine would first query for all 1000 and then set it

Not is it's set to extra lazy: http://docs.doctrine-project.org/projects/doctrine-orm/en/latest/tutorials/extra-lazy-associations.html

Reply:
So... this is tricky. If we generate code that has fetch="EXTRA_LAZY", then setting the inverse side, only costs one "fast" to check of the item already is related. If we do not set "EXTRA_LAZY", then it is far too costly.

This leaves us with 2 choices:
1) Generate all relationships as EXTRA_LAZY, sync the inverse side (just for edge-case convenience) and suffer a small perf impact

2) Do no generate relationships as EXTRA_LAZY, do not sync the inverse side (edge case problems could arise) and suffer no per impact.

I'm really not convinced either way. @stof @Majkl578 you usually have words of warning - what do you have here? Btw, we now ask the user IF they want to generate the inverse side. If they choose no, we don't have to worry about any of this.

I lean slightly towards option (1). EXTRA_LAZY seems better in 99% of the cases. And I don't like the small perf impact, but it really is small, and avoids WTF situations. I can't think of a situation where this causes a BIG perf impact.

Reply:
> EXTRA_LAZY seems better in 99% of the cases.

That's not actually true. If your business need means that the collection will never become huge, performing DB queries each time you check `contains` or `count` or similar APIs on the collection is actually much worse for performance than loading the collection on first usage (trading a bit of memory to win on time). I actually reverted a bunch of my extra lazy collections to be only lazy ones (i.e. the default) to solve performance issues.

For me, the right fix is to stop generating bidirectional relations by default. All this comes from the fact that the code expects a fully synchronized bi-directional relation. If you make the relation unidirectional, you make all these issues disappear (if you try to load the list of all articles before saving the change the DB, you might have a harder time writing your listener indeed as there is *no* place having the source of truth for the updated full list yet, but then it might push you to reconsider this need instead of killing performance and making the entity code very complex to have this list available).

Reply:
I thought more about this too and had the same thought as you Stof - that EXTRA_LAZY could cause tons of count queries in certain situations.

> If you make the relation unidirectional, you make all these issues disappear

It's true. But you've also killed a lot of usability for many users. To strike a balance, the command now DOES ask if you want the inverse side to be mapped. So, it's up to you